src/libponyc/expr/reference.c

```diff
@@ -168,10 +168,20 @@ bool expr_param(pass_opt_t* opt, ast_t* ast)
 
     type = consume_type(type, TK_NONE);
     errorframe_t err = NULL;
+    errorframe_t err2 = NULL;
 
-    if(!is_subtype(init_type, type, &err, opt))
+    if(type == NULL)
+    {
+      // other checks should have blocked us already
+      //pony_assert(0);
```

Comment:
Please ensure this line eventually gets either uncommented or removed.

Reply:
im confused by the assert and the meaning of the comment, is this supposed to never happen?

Reply:
Yes this is never supposed to happen as the code should have been rejected before reaching here. But I'm not certain whether it's actually before or just simultaneous. We might want it to behave defensively instead and use the pre-consumed type.

Reply:
Ok, so I would say something like...

"This should never happen" as the comment before the assert. It seems reasonable to me to add a comment that says something like 

"This should never happen. I've left this assert here for now because I'm not sure it won't happen. If enough time passes and this hasn't triggered, we can remove it."

Reply:
The second is reasonable